### PR TITLE
GEODE-1897: Add eviction option to create region

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionFunctionArgs.java [hunk 1]
```diff
@@ -128,6 +132,24 @@ public void setRegionExpirationTTL(Integer timeout, String action) {
...
 
+  public void setEvictionAttributes(String action, Integer maxMemory, Integer maxEntryCount) {
+    if (action == null) {
+      return;
+    }
+
+    EvictionAction evictionAction = EvictionAction.parseAction(action);
+    if (maxMemory == null && maxEntryCount == null) {
+      evictionAttributes =
+          EvictionAttributes.createLRUHeapAttributes(ObjectSizer.DEFAULT, evictionAction);
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionFunctionArgs.java]: Users are supposed to be able to plugin their own implementation of ObjectSizer. So you should probably support an "eviction-sizer" that takes a class name of the ObjectSizer they want to use for heap or mem lru.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java [hunk 2]
```diff
@@ -861,6 +861,15 @@
...
       "Action to be taken on a region that has exceeded the TTL expiration.";
+  public static final String CREATE_REGION__EVICTION_ACTION = "eviction-action";
+  public static final String CREATE_REGION__EVICTION_ACTION__HELP =
+      "The eviction action to apply. Must be either 'destroy' or 'overflow-to-disk'";
+  public static final String CREATE_REGION__EVICTION_MAX_MEMORY = "eviction-max-memory";
+  public static final String CREATE_REGION__EVICTION_MAX_MEMORY__HELP =
+      "Activates LRU eviction based on the region's memory usage specified by this value.";
+  public static final String CREATE_REGION__EVICTION_ENTRY_COUNT = "eviction-entry-count";
+  public static final String CREATE_REGION__EVICTION_ENTRY_COUNT__HELP =
+      "Activates LRU eviction based on the region's entry count specified by this value.";
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: When do we stop adding to CliStrings and just define these new strings where they're used in the Command classes? (See the email discussion thread " [Discuss] CliStrings".)

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java [hunk 3]
```diff
@@ -861,6 +861,15 @@
       "region-time-to-live-expiration-action";
   public static final String CREATE_REGION__REGIONEXPIRATIONTTLACTION__HELP =
       "Action to be taken on a region that has exceeded the TTL expiration.";
+  public static final String CREATE_REGION__EVICTION_ACTION = "eviction-action";
+  public static final String CREATE_REGION__EVICTION_ACTION__HELP =
+      "The eviction action to apply. Must be either 'destroy' or 'overflow-to-disk'";
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: The EvictionAction.stringTable defined is "local-destroy", not "destroy". COmmnet also applies to the CREATE_REGION__MSG__INVALID_EVICTION_ACTION string.

Comment (jdeppe-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: That is correct. The story calls for the eviction action to be `destroy` which is how it is exposed to the user via gfsh. The internal naming remains unchanged - i.e. `local-destroy`. I will double-check to ensure this is the correct requirement.

Comment (jdeppe-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: Yes, you're right. The current implementation does have the action as `local-destroy`. I have another PR in the wings which switches `EvictionAction` to an enum and that was what I was thinking of. When that hits, the eviction action option will be `destroy`.

**Pull request merged**